crates/polars-stream/src/nodes/io_sources/ipc.rs

```diff
@@ -0,0 +1,556 @@
+                        let BatchMessage {
+                            memslice: source,
+                            metadata,
+                            file_path,
+                            row_idx_offset,
+                            slice,
+                            seq,
+                            block_range,
+                        } = m;
+
+
+                        let mut reader = FileReader::new_with_projection_info(
+                            Cursor::new(source.as_ref()),
+                            metadata.as_ref().clone(),
+                            std::mem::take(&mut projection_info),
+                            None,
+                        );
+                        reader.set_current_block(block_range.start);
+                        reader.set_scratches((
+                            std::mem::take(&mut data_scratch),
+                            std::mem::take(&mut message_scratch),
+                        ));
+
+                        // Create the DataFrame with the appropriate schema and append all the record
+                        // batches to it. This will perform schema validation as well.
+                        let mut df = DataFrame::empty_with_schema(&pl_schema);
+                        df.try_extend(reader.by_ref().take(block_range.len()))?;
+
+                        df = df.slice(slice.start as i64, slice.len());
+
+                        if config.rechunk {
+                            df.rechunk_mut();
+                        }
+
+                        if let Some(RowIndex { name, offset: _ }) = &config.row_index {
+                            let offset = row_idx_offset + slice.start as IdxSize;
+                            df = df.with_row_index(name.clone(), Some(offset))?;
+                        }
+
+                        if let Some(col) = config.include_file_paths.as_ref() {
+                            let file_path = file_path.unwrap();
+                            let file_path = Scalar::from(PlSmallStr::from(file_path.as_ref()));
+                            df.with_column(Column::new_scalar(
+                                col.clone(),
+                                file_path,
+                                df.height(),
+                            ))?;
+                        }
+
+                        // If the block is very large, we want to split the block amongst the
+                        // pipelines. That will at least allow some parallelism.
+                        if df.height() > max_morsel_size && config::verbose() {
+                            eprintln!("IPC source encountered a (too) large record batch of {} rows. Splitting and continuing.", df.height());
+                        }
+                        for i in 0..df.height().div_ceil(max_morsel_size) {
+                            let morsel = df.slice((i * max_morsel_size) as i64, max_morsel_size);
+                            let seq = MorselSeq::new(seq + i as u64);
```

Comment:
I think we should initialize a `MorselSeq` at the start and use `MorselSeq::successor()` to increment this (it doesn't increment by 1)